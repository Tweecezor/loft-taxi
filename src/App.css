*{
    margin:0;
    padding:0;
    box-sizing: border-box;
}
ul{
    list-style: none;
}
a{
    text-decoration: none;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}
.container{
    width:1170px;
    margin:0 auto;
}
/* .container--map{
    position: relative;
} */
.header-wrap{
    box-shadow: 0 1px 4px 1px rgba(0, 0, 0, 0.11);
    border-bottom: 1px solid rgba(0,0,0,0.11);
    position: absolute;
    /* top: 0; */
    /* left: 0; */
    background: white;
    width: 100%;
    z-index:10;
}
.map{
    height:100vh;
    position: relative;
}

/* .background::before{
    content:'';
    position: absolute;
    width:100%;
    height: 100%;
    background: rgba(0,0,0,0.3);
    z-index:-1;
} */
.root-wrapper{
    position: relative;
}
input{
    border:none;
    border-bottom: 1px solid #eaeaea;
    padding-bottom: 10px;
    margin-bottom: 10px;
}
/* .registration__logo{
    margin-right: 50px;
} */
/* #map {
    position:absolute; top:0; bottom:0; width:100%;
} */

