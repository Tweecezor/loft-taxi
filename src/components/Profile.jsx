import React,{useState,useEffect} from 'react';
import '../styles/Profile.css';
import { connect } from 'react-redux';
import { sendProfileDataRequest,getProfileDataRequest,getCardDataRequest,changeCardNumber,changeCardData,changeCardOwner,changeCardCVC } from '../Actions/actions'
import {getCardsData,getUserToken} from '../reducers';

var styles = {
    background:'white',
    width:'800px',
    padding: '30px 35px'

}

const Profile = (props) => {

    var flag = 1;
    console.log(props.isLoading);

    useEffect(()=>{
        console.log(props)
       props.getCardData(props.userToken);
    },[flag]);
    //var
    // var [cardNumber,setCardNumber] = React.useState(props.cardsData.cardNumber);
    // var [cardData,setCardData] = React.useState(props.cardsData.cardData);
    // var [cardOwner,setCardOwner] = React.useState(props.cardsData.cardOwner);
    // var [cardCVC,setCardCVC] = React.useState(props.cardsData.cardCVC);
    var cardNumber = props.cardsData.cardNumber;
    var cardData = props.cardsData.cardData;
    var cardOwner = props.cardsData.cardOwner;
    var cardCVC = props.cardsData.cardCVC;

    const handleInputCardNumber = (e) => {
        props.setCardNumber(e.target.value);
    }
    const handleInputCardData = (e) => {
        props.setCardData(e.target.value);
    }
    const handleInputCardOwner = (e) => {
        props.setCardOwner(e.target.value);
    }
    const handleInputCardCVC = (e) => {
        props.setCardCVC( e.target.value);
    }

    const onSubmitForm = (e) => {
        e.preventDefault();
        console.log(props.cardsData);
        props.addCardsData(props.cardsData,props.userToken)
    }



    return (
        <div className = 'background'> 
            <div className = 'profile' style = {styles}>
                <div className = 'profile__header-wrap'>
                    <h2>Профиль</h2>
                    <h4>Способ оплаты</h4>
                </div>
                <form action="" className='profile__form' onSubmit = {onSubmitForm}>
                    <div className = 'profile__info-wrapper'>
                        <div className = 'profile__info'>
                            <div className = 'profile__info-wrap'>
                                <label className = 'profile__info-label' htmlFor = "name">Номер карты*</label>
                                <input className = 'profile__info-input' value = {cardNumber} type = "text" name = 'cardNumber' id = 'cardNumb' placeholder = '0000 0000 0000 0000' onChange = {handleInputCardNumber}/>
                            </div>
                            <div className = 'profile__info-wrap'>
                                <label className = 'profile__info-label' htmlFor = "name">Срок карты</label>
                                <input className = 'profile__info-input' value = {cardData}  type = "text" name = 'cardData' id = 'cardData' onChange = {handleInputCardData}/>
                            </div>
                        </div>
                        <div className = 'profile__info'>
                            <div className = 'profile__info-wrap'>
                                <label className = 'profile__info-label' htmlFor = "name">Имя владельца*</label>
                                <input className = 'profile__info-input' value = {cardOwner}  type = "text" name = 'email' id = 'name' placeholder = 'Petr Petrov' onChange = {handleInputCardOwner}/>
                            </div>
                            <div className = 'profile__info-wrap'>
                                <label className = 'profile__info-label' htmlFor = "name">CVC*</label>
                                <input className = 'profile__info-input' value = {cardCVC}  type = "text" name = 'cvc' id = 'cvc' onChange = {handleInputCardCVC}/>
                            </div>
                        </div>
                    </div>
                    <input type="submit" value = 'Сохранить' className = 'profile__form-submit'/>
                </form>
            </div>
        </div>
    )
}
const mapStateToProps = (state) => {
    return({
        cardsData:getCardsData(state),
        userToken:getUserToken(state),
        isLoading:state.ProfileReducer.isLoading
    })
}
const mapDispatchToProps = (dispatch) => {
    // console.log(dispatch)
    return ({
        addCardsData : (cardsData,token)=> {
            console.log('worked');
            dispatch(sendProfileDataRequest({...cardsData,token}));
        },
        getCardData : (userToken)=>{
            // debugger;
            dispatch(getCardDataRequest({userToken}))
        },
        setCardNumber : (cardNumber)=>{
            dispatch(changeCardNumber({cardNumber}))
        },
        setCardData : (cardData)=>{
            dispatch(changeCardData({cardData}))
        },
        setCardOwner : (cardOwner)=>{
            dispatch(changeCardOwner({cardOwner}))
        },
        setCardCVC : (cardCVC)=>{
            dispatch(changeCardCVC({cardCVC}))
        }

    })
}


export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Profile);