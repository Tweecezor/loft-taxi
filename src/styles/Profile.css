.profile__form{
    /* display: flex;
    flex-direction: column; */
    text-align: center;
}
.profile__info-wrap{
    display: flex;
    flex-direction: column;
    width:100%
}
.profile__info-label{
    margin-bottom: 10px;
}
.profile__info-input{
   margin-bottom: 10px;
}
.profile__form-submit{
    padding: 15px 20px;
    background-color: #ffc617;
    border-radius: 4px;
    text-transform: uppercase;
    font-weight: 800;
    border:none;
}
.profile__info{
    border:1px solid black;
    width: 300px;
    padding:20px;
    border-radius: 5px;
}
.profile{
    display: flex;
    flex-direction: column;
}
.profile__header-wrap{
    text-align: center;
    margin-bottom: 30px;
}
.profile__info-wrapper{
    display: flex;
    justify-content: space-between;
    margin-bottom: 30px;
}
.profile__header-wrap > h2{
    font-size: 40px;
}
.profile__header-wrap > h4{
    font-size: 20px;
    font-weight: normal;
}