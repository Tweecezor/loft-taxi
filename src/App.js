import React,{useState} from 'react';

import './App.css';
import Header from './components/Header';

import Map from './components/Map';

import Profile from './components/Profile';
import Background from './components/Background';

import myContext from '../src/context' ;


var currentBlock = 'login';
const {Provider,Consumer} = myContext;






const PAGES = {
    map : () => (    
        <Map />
        ) ,
    profile : () => (
        <section className = 'background'>
            <Profile />
        </section>
    ),
    // welcome : (setPage) => (
       
    //         <Background setPageProp = {setPage} />
      
    // )
}

function App() {

    const login = (email, password) => {
        if (email === 'test' && password === '123') {
            setIsLoggedIn(true)
            // setPage('profile')
        }
    }
    const logout = () => {
        setIsLoggedIn(false)
        // setPage('login')
    }
   
    const [isLoggedIn,setIsLoggedIn] = React.useState(false);
    const [page,setPage] = React.useState('map');
    return (
        <Provider value = { {isLoggedIn,setIsLoggedIn,login,logout}  }>
            {isLoggedIn &&(
                 <div className = 'root-wrapper'>
                    <Header setPageProp = {setPage} />
                    {PAGES[page](setPage)}
                </div>
            )}
            {!isLoggedIn && (
                <Background setPageProp = {setPage} />
            )}
          
           
           
        </Provider>
    );


}

export default App;






// пример как работает код выше
// class newclass extends React.Component {

//     state = {
//         count : 1 
//     }
//     handleClick = ()=>{
//         this.setState({count: this.state.count++})
//     }

//     render(){
//         return(
//             <CompMy onClick={this.handleClick}/>        )
//     }
// }
