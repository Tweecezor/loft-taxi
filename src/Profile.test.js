import React,{useState} from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow,mount} from 'enzyme';
import Profile from './components/Profile';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';


it('Profile Snapshot', () => {
    const tree = renderer.create(<Profile/>).toJSON();
    expect(tree).toMatchSnapshot();
})