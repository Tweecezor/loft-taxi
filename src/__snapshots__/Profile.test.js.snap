// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Snapshot 1`] = `
<div
  className="profile"
  style={
    Object {
      "background": "white",
      "padding": "90px 35px",
      "width": "450px",
    }
  }
>
  <h2>
    Profile
  </h2>
</div>
`;
